### PR TITLE
ROC-4573 Reject Settlement Agreement [Defendant] 

### PR DESCRIPTION
**Pull request by ellie-harrison**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4573

### Change description
Add journey to reject a settlement agreement from the claimant side

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/dashboard/views/macro/claimStatus/settlementAgreementClaimant.njk [hunk 1]
```diff
@@ -62,6 +66,18 @@
   </div>
 {% endmacro %}
 
+{% macro settlementAgreementRejectedClaimantClaimDetails(claim, csrf) %}
+    <h2 class="heading-small">{{ t('The defendant has rejected your settlement agreement') }}</h2>
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/settlementAgreementClaimant.njk]: this doesn't match the style of the other statuses. Have asked ronan to update the prototype so this macro and the defendant version may need to be updated once he's reviewed the prototype and made changes

File: src/main/features/dashboard/views/macro/claimStatus/settlementAgreementDefendant.njk [hunk 2]
```diff
@@ -61,7 +65,21 @@
     <p>{{ t('They asked you to sign a settlement agreement to formalise the plan.') }}</p>
     <p>{{ t('If you sign the agreement, they can’t request a County Court Judgment against you unless you break the terms.') }}</p>
     <p>{{ t('If you don’t sign or respond by {{ deadline }} they can request a County Court Judgment against you.', deadline = claim.claimantRespondedAt | addDays(7) | date) }}</p>
-    <p>{{ internalLink('View the repayment plan', CCJPaths.repaymentPlanSummaryPage.evaluateUri({ externalId: claim.externalId, madeBy: MadeBy.DEFENDANT.value })) }}</p>
+    <p>{{ internalLink('View the repayment plan', SettlementAgreementPaths.signSettlementAgreement.evaluateUri({ externalId: claim.externalId })) }} </p>
+        <!--CCJPaths.repaymentPlanSummaryPage.evaluateUri({ externalId: claim.externalId, madeBy: MadeBy.CLAIMANT.value })) }}</p>-->
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/settlementAgreementDefendant.njk]: can this be deleted?

File: src/main/features/dashboard/views/macro/claimStatus/settlementAgreementRejected.njk [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{% macro offerRejectedForClaimantDashboard(claim) %}
+  <p>
+    {{ t('You’ve rejected an offer to settle out of court.') }}
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/settlementAgreementRejected.njk]: Please put this all on one line. adding line breaks between p tags can add white space to the content which can cause weird display effects.

File: src/main/features/dashboard/views/macro/claimStatus/settlementAgreementRejected.njk [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    {% set defendantName = claim.claimData.defendant.name %}
+    <h2 class="heading-small">{{ t('Settle out of court') }}</h2>
+    <p>{{ t('You’ve rejected the defendant’s offer to settle out of court. '
+        + 'You won’t receive any more offers from the defendant.') }}</p>
+  </div>
+{% endmacro %}
+
+{% macro offerRejectedForDefendantDashboard() %}
+  <p>
+    {{ t('The claimant has rejected your offer.') }}
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/settlementAgreementRejected.njk]: as above

File: src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{% extends "layout.njk" %}
+{% from "support.njk" import supportWidget %}
+{% from "externalLink.njk" import externalLink %}
+
+{% extends "layout.njk" %}
+
+{% block content %}
+<div class="grid-row">
+  <div class="column-two-thirds">
+
```
Comment (robertparkinson) [inline src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk]: lots of empty lines for no reason. Can we delete a lot of them to make the code more compact?

File: src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      <p>{{ t('The claimant can request a County Court Judgment (CCJ) against you. A CCJ would order you to repay the money in line with the terms of the agreement.') }}</p>
+      <p>{{ t('If they request a CCJ, you can ask a judge to consider changing the repayment plan, based on your financial details.') }}</p>
+      <p>{{ t('The court has reviewed the plan and believes you can afford it, so a judge may not change it.') }}</p>
+      <p>{{ t('We’ll email you when the claimant responds.') }}</p>
+    {% endif %}
+
+
+    {% include "includes/service_survey_link.njk" %}
+
+    <p><a class="button" href="{{ DashboardPaths.dashboardPage.uri }}">{{ t('Go to your account') }}</a></p>
```
Comment (robertparkinson) [inline src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk]: switch to internal link and remove the p. if it needs padding add a div with a form-group

File: src/main/features/settlement-agreement/views/sign-settlement-agreement.njk [hunk 7]
```diff
@@ -0,0 +1,66 @@
+{% from "form.njk" import csrfProtection, radioGroup, saveAndContinueButton, errorSummary %}
+{% from "table.njk" import tableStart, tableEnd, row, rowWithMultipleValue %}
+{% from "./macro/settlement-agreement.njk" import settlementAgreementSummary %}
+
+{% extends "layout.njk" %}
+
+{% set heading = t('Respond to the settlement agreement') %}
+
+{% block content %}
+
```
Comment (robertparkinson) [inline src/main/features/settlement-agreement/views/sign-settlement-agreement.njk]: again lots of empty lines for no reason. please clean up.

File: src/test/features/settlement-agreement/form/models/defendantSettlementResponse.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import { expect } from 'chai'
+import { Validator } from 'class-validator'
+import { expectValidationError } from 'test/app/forms/models/validationUtils'
+import {
+  DefendantSettlementResponse,
+  DefendantSettlementResponseOption,
+  ValidationErrors
+} from '../../../../../main/features/settlement-agreement/form/models/defendantSettlementResponse'
```
Comment (robertparkinson) [inline src/test/features/settlement-agreement/form/models/defendantSettlementResponse.ts]: please replace ../../..

File: src/main/app/claims/models/claim.ts [hunk 9]
```diff
@@ -254,6 +257,16 @@ export class Claim {
     return this.settlement && this.settlement.isThroughAdmissionsAndSettled()
   }
 
+  private isSettlementAgreementRejected (): boolean {
+    if (this.claimantResponse && this.claimantResponse.type === ClaimantResponseType.ACCEPTATION) {
+      const claimantResponse: AcceptationClaimantResponse = this.claimantResponse
+      return claimantResponse.formaliseOption === FormaliseOption.SETTLEMENT
+        && this.settlement && this.settlement.isOfferRejected()
+    } else {
+      return false
```
Comment (andrewwa-kainos) [inline src/main/app/claims/models/claim.ts]: minor: readability proposal
```
private isSettlementAgreementRejected (): boolean {
    if (!this.claimantResponse || this.claimantResponse.type !== ClaimantResponseType.ACCEPTATION) {
        return false
    }
    const claimantResponse: AcceptationClaimantResponse = this.claimantResponse
    return claimantResponse.formaliseOption === FormaliseOption.SETTLEMENT
        && this.settlement && this.settlement.isOfferRejected()
}
```

File: src/main/features/dashboard/views/macro/claimStatus/settlementAgreementDefendant.njk [hunk 10]
```diff
@@ -61,7 +65,20 @@
     <p>{{ t('They asked you to sign a settlement agreement to formalise the plan.') }}</p>
     <p>{{ t('If you sign the agreement, they can’t request a County Court Judgment against you unless you break the terms.') }}</p>
     <p>{{ t('If you don’t sign or respond by {{ deadline }} they can request a County Court Judgment against you.', deadline = claim.claimantRespondedAt | addDays(7) | date) }}</p>
-    <p>{{ internalLink('View the repayment plan', CCJPaths.repaymentPlanSummaryPage.evaluateUri({ externalId: claim.externalId, madeBy: MadeBy.DEFENDANT.value })) }}</p>
+    <p>{{ internalLink('View the repayment plan', SettlementAgreementPaths.repaymentPlanSummary.evaluateUri({ externalId: claim.externalId })) }}</p>
```
Comment (andrewwa-kainos) [inline src/main/features/dashboard/views/macro/claimStatus/settlementAgreementDefendant.njk]: Filter `safe`?

File: src/main/features/dashboard/views/macro/claimStatus/utilityMacros.njk [hunk 11]
```diff
@@ -8,7 +8,7 @@
 
 {% macro signAgreementLink(claim, text = 'Sign the settlement agreement') %}
   {% if claim.settlement.isThroughAdmissions() %}
-    <a href="{{ OfferPaths.declarationPage.evaluateUri({ externalId: claim.externalId }) }}"
+    <a href="{{ SettlementAgreementPaths.signSettlementAgreement.evaluateUri({ externalId: claim.externalId }) }}"
```
Comment (andrewwa-kainos) [inline src/main/features/dashboard/views/macro/claimStatus/utilityMacros.njk]: Can be an `internalLink`

File: src/main/features/settlement-agreement/guards/settlementAgreementGuard.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+  static async requestHandler (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+    const claim: Claim = res.locals.claim
+    const settlement: Settlement = claim.settlement
+    const claimantResponse: ClaimantResponse = claim.claimantResponse
+
+    if (!claimantResponse || claimantResponse.type !== ClaimantResponseType.ACCEPTATION
+      || AcceptationClaimantResponse.deserialize(claimantResponse).formaliseOption !== FormaliseOption.SETTLEMENT) {
+      logger.warn(`Claim ${claim.claimNumber} no acceptance claimant response for claim`)
+    } else if (!settlement || settlement.isSettled() || settlement.isOfferRejected()) {
```
Comment (andrewwa-kainos) [inline src/main/features/settlement-agreement/guards/settlementAgreementGuard.ts]: The first branch of code will log and then not do anything - neither redirect nor `next()`

File: src/main/features/settlement-agreement/routes/repayment-plan-summary.ts [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+import { Form } from 'main/app/forms/form'
+import { PaidAmount } from 'ccj/form/models/paidAmount'
+import { Claim } from 'main/app/claims/models/claim'
+import { PaymentIntention } from 'main/app/claims/models/response/core/paymentIntention'
+
+function renderView (form: Form<PaidAmount>, req: express.Request, res: express.Response): void {
+  const claim: Claim = res.locals.claim
+  let paymentIntention: PaymentIntention
+
+  // TODO: This needs to be a guard
```
Comment (andrewwa-kainos) [inline src/main/features/settlement-agreement/routes/repayment-plan-summary.ts]: Can you action the TODO already?

File: src/main/features/settlement-agreement/views/repayment-plan-summary.njk [hunk 14]
```diff
@@ -0,0 +1,38 @@
+{% extends "layout.njk" %}
+{% from "internalLink.njk" import internalLink %}
+{% from "form.njk" import csrfProtection, submitButton %}
+{% from "paymentPlanTable.njk" import paymentPlanTable %}
+
+  {% set heading = 'The claimant’s repayment plan' %}
+
```
Comment (andrewwa-kainos) [inline src/main/features/settlement-agreement/views/repayment-plan-summary.njk]: minor: extra whitespace

File: src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+{% from "externalLink.njk" import externalLink %}
+{% from "internalLink.njk" import internalLink %}
+
+{% extends "layout.njk" %}
+
+{% block content %}
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <div class="govuk-box-highlight">
+      {% if claim.settlement.isOfferRejected() %}
```
[the remark this answers is not in the document]
Comment (ellie-harrison) [reply src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk]: There is missing content for the Acceptation part of the journey, this stuff will be added as part of ROC-4847

File: src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+        <h1 class="bold-large">{{ t('You’ve rejected the settlement agreement') }}</h1>
+      {% endif %}
+    </div>
+    <h2 class="heading-medium">{{ t('What happens next') }}</h2>
+    {% if claim.settlement.isOfferRejected() %}
```
[the remark this answers is not in the document]
Comment (ellie-harrison) [reply src/main/features/settlement-agreement/views/settlement-agreement-confirmation.njk]: As mentioned previously, there is an acceptation part of this journey that will be added in ROC-4847

**Pull request merged**